test/extended/networking/livemigration.go

```diff
@@ -627,3 +823,33 @@ func networkName(netSpecConfig string) string {
 	Expect(json.Unmarshal([]byte(netSpecConfig), &nc)).To(Succeed())
 	return nc.Name
 }
+
+// formatAddressesAnnotation converts slice of IPs to the required JSON format for kubevirt addresses annotation
+func formatAddressesAnnotation(preconfiguredIPs []string) (string, error) {
+	const primaryUDNNetworkName = "overlay"
+	if len(preconfiguredIPs) == 0 {
+		return "", nil
+	}
+
+	ips := make([]string, len(preconfiguredIPs))
+	for i, ip := range preconfiguredIPs {
+		ips[i] = strings.TrimSpace(ip)
+	}
+
+	staticIPs, err := json.Marshal(map[string][]string{
+		primaryUDNNetworkName: ips,
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal static IPs: %w", err)
+	}
+
+	return string(staticIPs), nil
+}
+
+func ipRequests(ips string) map[string]string {
+	return map[string]string{kvIPRequestsAnnot: ips}
+}
+
+func hasMACAddressRequest(vmiSpecJSON string) bool {
+	return strings.Contains(vmiSpecJSON, "macAddress")
+}
```

Comment:
@RamLavi yeap this part here sucks (... not saying it's the only place that does ...). Please work on top of this current iteration to improve this.